src/make_android.sh

```diff
@@ -70,7 +70,62 @@ rm -rf *
 cd ..
 
 echo --- Copying content
+for i in $*; do 
+  if [ ${i%%=*} = "exclude-activities" ]
+  then 
+    activities=${i#*=}
+    lastActivity=${activities##*,}
+    remaining=true
+    echo -e "\n" >> exclude.android
+    cp ../sugarizer/activities.json ../sugarizer/activities.bak.json
+    while [ $remaining == true ]
+    do
+      activity=${activities%%,*}
+      activities=${activities#*,}
+      if [ ${#activity} -gt 0 ]
+      then
+        sed -i "/${activity}/d" ../sugarizer/activities.json
+        echo "activities/$activity.activity/" >> exclude.android
+      fi
+      if [ $activity = $lastActivity ]
+      then
+        remaining=false
+      fi
+    done
+    sed -i "$(( $( wc -l < ../sugarizer/activities.json) ))s/,$//" ../sugarizer/activities.json
+  elif [ ${i%%=*} = "include-activities" ]
+  then
+    touch include.android
+    activities=${i#*=}
+    lastActivity=${activities##*,}
+    remaining=true
+    mv ../sugarizer/activities.json ../sugarizer/activities.bak.json
+    touch ../sugarizer/activities.json
+    echo -e "[" >> ../sugarizer/activities.json
+    while [ $remaining == true ]
+    do
+      activity=${activities%%,*}
+      activities=${activities#*,}
+      if [ ${#activity} -gt 0 ]
+      then
+        grep ${activity} ../sugarizer/activities.bak.json >> ../sugarizer/activities.json
+        echo "activities/$activity.activity/" >> include.android
+      fi
+      if [ $activity = $lastActivity ]
+      then
+        remaining=false
+      fi
+    done
+    echo -e "]" >> ../sugarizer/activities.json
+    sed -i "$(( $( wc -l < ../sugarizer/activities.json) -1 ))s/,$//" ../sugarizer/activities.json
+  fi
+done
+
 rsync -av --exclude-from='exclude.android' ../sugarizer/* www
+
+rm ../sugarizer/activities.json
+mv ../sugarizer/activities.bak.json ../sugarizer/activities.json
+
 cp etoys_remote.index.html www/activities/Etoys.activity/index.html
 if [ "$1" == "minsize" -o "$2" == "minsize" -o "$3" == "minsize" -o "$4" == "minsize" ]; then
```

Comment:
If you add one argument, I think you should update also the way the script retrieves following options: `minsize`, `full`, `release`, `sign`, `os`.
Because to test these options, the script just do a positional comparison. So, for example here, the `minsize` option could be located as `$5` and will be ignore if the `exclude-activities` option is here.

Reply:
I agree, should I implement a `for` loop for these operations as well? It would be better than explicitly comparing all arguments and would work even if a new argument is added in the future.

Reply:
Yes, will be great to retrieve all arguments at the same place.

Reply:
Due to the synchronous nature of the builder, we can't perform all tasks inside a single `for` loop. Therefore I have added simple variable flags for each argument and used these flags all over the code. This allows for a central argument retrieval using only 1 loop.